src/Cli/dotnet/commands/dotnet-test/MSBuildHandler.cs

```diff
@@ -0,0 +1,269 @@
+            string targetFrameworks = project.GetPropertyValue(ProjectProperties.TargetFrameworks);
+            string targetPath = project.GetPropertyValue(ProjectProperties.TargetPath);
+            string projectFullPath = project.GetPropertyValue(ProjectProperties.ProjectFullPath);
+            string runSettingsFilePath = project.GetPropertyValue(ProjectProperties.RunSettingsFilePath);
+
+            var projects = new List<Module>();
+
+            if (string.IsNullOrEmpty(targetFrameworks))
+            {
+                projects.Add(new Module(targetPath, projectFullPath, targetFramework, runSettingsFilePath, isTestingPlatformApplication, isTestProject));
+            }
+            else
+            {
+                var frameworks = targetFrameworks.Split(Separator, StringSplitOptions.RemoveEmptyEntries);
+                foreach (var framework in frameworks)
+                {
+                    project.SetProperty(ProjectProperties.TargetFramework, framework);
+                    project.ReevaluateIfNecessary();
+
+                    projects.Add(new Module(project.GetPropertyValue(ProjectProperties.TargetPath),
+                        projectFullPath,
+                        framework,
+                        runSettingsFilePath,
+                        isTestingPlatformApplication,
+                        isTestProject));
+                }
+            }
+
+            return projects;
+        }
+
+        private static BuildResult RestoreProject(string projectFilePath, ProjectCollection projectCollection, bool allowBinLog, string binLogFileName)
+        {
+            BuildParameters parameters = new(projectCollection)
+            {
+                Loggers = [new ConsoleLogger(LoggerVerbosity.Quiet)]
+            };
+
+            if (allowBinLog)
+            {
+                parameters.Loggers = parameters.Loggers.Concat([
+                    new BinaryLogger
+                    {
+                        Parameters = binLogFileName
+                    }
+                ]);
+            }
+
+            var buildRequestData = new BuildRequestData(projectFilePath, new Dictionary<string, string>(), null, [CliConstants.RestoreCommand], null);
+            BuildResult buildResult;
+            lock (buildLock)
+            {
+                buildResult = BuildManager.DefaultBuildManager.Build(parameters, buildRequestData);
+            }
+
+            return buildResult;
+        }
+
+        private static bool IsBinaryLoggerEnabled(List<string> args, out string binLogFileName)
+        {
+            binLogFileName = BinLogFileName;
+
+            var binLogArgs = new List<string>();
+
+            foreach (var arg in args)
+            {
+                if (arg.StartsWith("/bl:") || arg.Equals("/bl")
+                    || arg.StartsWith("--binaryLogger:") || arg.Equals("--binaryLogger")
+                    || arg.StartsWith("-bl:") || arg.Equals("-bl"))
```

Comment:
I would probably have created an helper code shared with dotnet run to ensure both gets fixed but fine like this.